### DISCUSSION TITLE
xAlignment: 'left' for multi line text

### DISCUSSION THREAD

**Discussion by YongheeMinnovation** [clientIO/joint#1679] [Q&A] [answered]
Hi there,

I have a rectangle element with some text, and the text has three lines.
When I align the text to left, only one line is aligned and two lines are not aligned.
Is there any way I can align all lines together?

````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````
var rect = new joint.shapes.standard.Rectangle();
        rect.position(100, 30);
        rect.resize(70, 25);
        rect.attr({
            body: {
                fill: 'none',
                strokeWidth: 0
            },
            label: {
                text: inputText + '\n' + title + '\n' + value,
                xAlignment: 'left',
                fontSize: 18
            }
        });
        rect.addPort(port);
        rect.addTo(graph);
````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````
![Capture](https://user-images.githubusercontent.com/93103266/170416871-86f53bff-3f96-4d83-84dd-e1389bd89493.JPG)

Thank you so much!

**Comment by kumilingus** [comment, marked as answer]
Use the SVG `textAnchor` attribute ([MDN documentation](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/text-anchor)) to align text **horizontally**.

```js
rect.attr({
    label: {
        text: inputText + '\n' + title + '\n' + value,
        textAnchor: 'start'
    }
});
```

And JointJS `textVerticalAnchor` custom attribute ([JointJS documentation](https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#dia.attributes.textVerticalAnchor)) to align text **vertically**.

```js
rect.attr({
    label: {
        text: inputText + '\n' + title + '\n' + value,
        x: 'calc(0.5 * w)',
        // position the text 10 pixels above the element
        y: -10,
        textVerticalAnchor: 'bottom'
    }
});
```

**Reply by YongheeMinnovation** [reply]
Awesome, It works perfectly!
Thank you so much.
